regression/boundary_conditions.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "generic_benchmark.hpp"
+
+#include <gridtools/boundary_conditions/boundary.hpp>
+#include <gridtools/tools/backend_select.hpp>
+#include <gridtools/tools/regression_fixture.hpp>
+
+#include <gtest/gtest.h>
+
+using namespace gridtools;
+
+namespace {
+    struct direction_bc_input {
+        float_type value;
+
+        GT_FUNCTION
+        direction_bc_input() : value(1) {}
+
+        GT_FUNCTION
+        direction_bc_input(float_type v) : value(v) {}
+
+        // relative coordinates
+        template <typename Direction, typename DataField0, typename DataField1>
+        GT_FUNCTION void operator()(
+            Direction, DataField0 &data_field0, DataField1 const &data_field1, uint_t i, uint_t j, uint_t k) const {
+            data_field1(i, j, k) = data_field0(i, j, k) * value;
+        }
+
+        // relative coordinates
+        template <sign I, sign K, typename DataField0, typename DataField1>
+        GT_FUNCTION void operator()(
+            direction<I, minus_, K>, DataField0 &, DataField1 const &data_field1, uint_t i, uint_t j, uint_t k) const {
+            data_field1(i, j, k) = 88 * value;
```

Comment:
Up to you: I don't like this hard-coded random numbers. Consider giving them a name. (I know it is copied from the old example...)

Reply:
done